auth-pam.c

```diff
@@ -576,16 +576,15 @@ sshpam_thread(void *ctxtp)
 	pthread_exit(NULL);
 
  auth_fail:
+	debug3("PAM: auth_fail sshpam_err=%d %s", sshpam_err, pam_strerror(sshpam_handle, sshpam_err));
 	if ((r = sshbuf_put_cstring(buffer,
 	    pam_strerror(sshpam_handle, sshpam_err))) != 0)
 		fatal("%s: buffer error: %s", __func__, ssh_err(r));
 	/* XXX - can't do much about an error here */
-	if (sshpam_err == PAM_ACCT_EXPIRED)
-		ssh_msg_send(ctxt->pam_csock, PAM_ACCT_EXPIRED, buffer);
-	else if (sshpam_maxtries_reached)
+	if (sshpam_maxtries_reached)
 		ssh_msg_send(ctxt->pam_csock, PAM_MAXTRIES, buffer);
 	else
-		ssh_msg_send(ctxt->pam_csock, PAM_AUTH_ERR, buffer);
+		ssh_msg_send(ctxt->pam_csock, sshpam_err, buffer);
```

Comment:
I don't understand this part of the change - AFAIK the `sshpam_err` value gets read back in `sshpam_query()` via `ssh_msg_recv()`, but that function would handle arbitrary error codes via its `default:` switch case - does this do what we want in all circumstances?

Maybe I'm not following (thanks for having largely swapped out the PAM code from my brain), but the material part of this change seems to be plumbing the correct error code back to the `sshbuf_put_cstring()` call above, and not this. Am I off-base here?